### PR TITLE
SNOW-1852779 Fix AST encoding for Column `in_`, `asc`, and `desc`

### PR DESCRIPTION
**Pull request by sfc-gh-vbudati**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1852779

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

3. Please describe how your code solves the related issue.

Fixed AST encoding for Column `in_`, `asc`, and `desc`. Removed an unused entity and renamed `sp_column_in__seq` to `sp_column_in`. Changed the `nulls_first` optional boolean param to be an enum.

### CODE REVIEW

**Review by sfc-gh-azwiegincew — APPROVED** [review]

File: src/snowflake/snowpark/column.py [hunk 1]
```diff
@@ -644,7 +644,7 @@ def in_(
             ast = None
             if _emit_ast:
                 ast = proto.Expr()
```
Comment (sfc-gh-lspiegelberg) [inline src/snowflake/snowpark/column.py:646]: is the IR server-side branch for this already merged?

File: tests/ast/data/windows.test [hunk 2]
```diff
@@ -557,6 +557,9 @@ body {
                                 }
                               }
                             }
```
Comment (sfc-gh-lspiegelberg) [inline tests/ast/data/windows.test:559]: Do we have a test for this where it is false?

File: src/snowflake/snowpark/column.py (hunk 1 above)
Comment (sfc-gh-vbudati) [reply src/snowflake/snowpark/column.py:646]: Yup!

File: tests/ast/data/windows.test (hunk 2 above)
Comment (sfc-gh-vbudati) [reply tests/ast/data/windows.test:559]: It can't be "false" -- it's an enum. It can only be set to true but this should be tested by the asc/desc tests.

**Pull request merged**